src/Doctrine/Instantiator/Instantiator.php

```diff
@@ -186,6 +187,10 @@ private function attemptInstantiationViaUnSerialization(ReflectionClass $reflect
             restore_error_handler();
 
             throw UnexpectedValueException::fromSerializationTriggeredException($reflectionClass, $exception);
+        } catch (Throwable $exception) {
+            restore_error_handler();
```

Comment:
I got your sentence backward (like the finally statement is allowed since 5.3/5.4)